packages/ra-ui-materialui/src/input/InPlaceEditor/InPlaceEditor.stories.tsx

```diff
@@ -0,0 +1,372 @@
+                                                        />
+                                                    }
+                                                >
+                                                    <NumberField source="age" />
+                                                </InPlaceEditor>
+                                            </DataTable.Col>
+                                            <DataTable.Col source="type">
+                                                <InPlaceEditor
+                                                    source="type"
+                                                    editor={
+                                                        <SelectInput
+                                                            source="type"
+                                                            choices={[
+                                                                {
+                                                                    id: 'prospect',
+                                                                    name: 'Prospect',
+                                                                },
+                                                                {
+                                                                    id: 'customer',
+                                                                    name: 'Customer',
+                                                                },
+                                                            ]}
+                                                            size="small"
+                                                            margin="none"
+                                                            label={false}
+                                                            variant="standard"
+                                                            autoFocus
+                                                            SelectProps={{
+                                                                defaultOpen:
+                                                                    true,
+                                                            }}
+                                                            helperText={false}
+                                                            sx={{
+                                                                width: 50,
+                                                                '& .MuiInputBase-root':
+                                                                    {
+                                                                        marginTop: 0,
+                                                                        fontSize: 14,
+                                                                    },
+                                                            }}
+                                                        />
```

Comment:
nitpick: this `SelectInput` should be `required`, because if you pick the empty option, then you can never edit it back (it renders an empty `div` so you can no longer click on it)